datafusion/src/logical_plan/expr.rs

```diff
@@ -1387,6 +1426,11 @@ macro_rules! make_literal {
             fn lit(&self) -> Expr {
                 Expr::Literal(ScalarValue::$SCALAR(Some(self.clone())))
             }
+
+            fn lit_timestamp_nanosecond(&self) -> Expr {
+                let scalar = ScalarValue::$SCALAR(Some(self.clone()));
+                scalar.lit_timestamp_nanosecond()
```

Comment:
I think this seems a little bit inefficient to wrap a numeric value with `ScalarValue` enum, then subsequently match on that enum to unpack the value and return a different enum. Perhaps we could find a way to execute `ScalarValue::TimestampNanosecond(Some(self.clone().into())` here directly?

Reply:
This seems a good solution to me👍

Reply:
+1

Reply:
I have tried that but not all data type works for the `into`